jsettlers.graphics/src/jsettlers/graphics/messages/Messenger.java

```diff
@@ -37,10 +49,42 @@
 		return messages;
 	}
 
-	public void addMessage(Message message) {
-		messages.addFirst(message);
-		if (messages.size() > MAX_MESSAGES) {
-			messages.removeLast();
+	public boolean addMessage(Message message) {
+		if (isNews(message)) {
+			messages.addFirst(message);
+			if (messages.size() > MAX_MESSAGES)
+				messages.removeLast();
+			return true;
+		}
+		return false;
+	}
+
+	public void removeOld() {
+		for (; (messages.size() > 0)
+				&& (messages.getLast().getAge()>MESSAGE_TTL);
+				messages.removeLast());
+	}
+
+	boolean isNews(Message msg) {
+		if (messages.isEmpty()) {
+			return true;
+		}
+		for (Message m : messages) {
+			if ((m.getSender() == msg.getSender())
```

Comment:
Can you move this to the message class?

Something like Message#isLike(otherMessage)

This would make the code cleaner.